plugins/utils/navbuddy.nix

```
@@ -0,0 +1,267 @@
+          For each keybinding it takes a table of format { callback = <function_to_be_called>, description = "string"}.
+          The callback function takes the "display" object as an argument.
+        '';
+        default = {
+          "<esc>" = "close";
+          "q" = "close";
+
+          "j" = "next_sibling";
+          "k" = "previous_sibling";
+
+          "h" = "parent";
+          "l" = "children";
+          "0" = "root";
+
+          "v" = "visual_name";
+          "V" = "visual_scope";
+
+          "y" = "yank_name";
+          "Y" = "yank_scope";
+
+          "i" = "insert_name";
+          "I" = "insert_scope";
+
+          "a" = "append_name";
+          "A" = "append_scope";
+
+          "r" = "rename";
+
+          "d" = "delete";
+
+          "f" = "fold_create";
+          "F" = "fold_delete";
+
+          "c" = "comment";
+
+          "<enter>" = "select";
+          "o" = "select";
+
+          "J" = "move_down";
+          "K" = "move_up";
+
+          "s" = "toggle_preview";
+
+          "<C-v>" = "vsplit";
+          "<C-s>" = "hsplit";
+        };
+      };
+
+      lsp = {
+        autoAttach = helpers.defaultNullOpts.mkBool false ''
+          If set to true, you don't need to manually use attach function
+        '';
+
+        preference = helpers.defaultNullOpts.mkListStr null ''
+          list of lsp server names in order of preference
+        '';
+      };
+
+      sourceBuffer = {
+        followNode = helpers.defaultNullOpts.mkBool true "Keep the current node in focus on the source buffer";
+        highlight = helpers.defaultNullOpts.mkBool true "Highlight the currently focused node";
+        reorient = helpers.defaultNullOpts.mkEnum [ "smart" "top" "mid" "none" ] "smart" ''
+          Right section can show previews too.
+                Options: "leaf", "always" or "never"
+        '';
+        scrolloff = helpers.defaultNullOpts.mkInt null ''
+          scrolloff value when navbuddy is open
+        '';
+      };
+    };
+
+  config =
+    let
+      setupOptions = with cfg;
+        {
+          inherit
+            window
+            icons
+            ;
+          node_markers = nodeMarkers;
+          use_default_mapping = useDefaultMapping;
+          lsp = with lsp; {
+            auto_attach = autoAttach;
+            inherit preference;
+          };
+          source_buffer = sourceBuffer;
+        }
+        // cfg.extraOptions;
+
+      mappings =
+        mapAttrs'
+          (action: key: {
+            name = key;
+            value = {
+              action = "function() require('nvim-navbuddy.actions').actions.${action}) end";
+              lua = true;
+              inherit (cfg.keymaps) silent;
+              #desc = "Spider-${motion}";
+            };
+          })
+          cfg.keymaps.motions;
+    in
+    mkIf cfg.enable {
+      extraPlugins = [cfg.package];
+
+      maps =
+        genAttrs
+          [ "normal" ]
+          (mode: mappings);
```

```suggestion
        genAttrs
        [ "normal" ]
        (mode: mappings);
```
Please, run `nix fmt && statix check . && nix flake check` before pushing to your branch.
Like so, we avoid formatting and basic errors :)

Thanks in advance !

---

okay will do :)